src/logic/currencyValues/store/middleware/index.ts

```diff
@@ -21,9 +22,9 @@ const currencyValuesStorageMiddleware = (store) => (next) => async (action) => {
       case SET_CURRENCY_RATE:
       case SET_CURRENCY_BALANCES: {
         const currencyValues = currencyValuesSelector(state)
-        const currencyValuesWithoutBalances = currencyValues.map((currencyValue) => {
-          const currencyRate = currencyValue.get('currencyRate')
-          const selectedCurrency = currencyValue.get('selectedCurrency')
+        const currencyValuesWithoutBalances: Map<string, CurrencyRateValue> = currencyValues.map((currencyValue) => {
```

Comment:
is currency rate a map or a record?
if it's a record, you can access its properties like this `currencyValue.currencyCare`

Reply:
it's a map :)

Reply:
shouldn't we use https://github.com/gajus/redux-immutable as is recommended in https://redux.js.org/recipes/using-immutablejs-with-redux#make-your-entire-redux-state-tree-an-immutablejs-object?

Reply:
I would personally get rid of immutable.js in favor of `redux-toolkit`, which is officially recommended by redux team now: https://redux.js.org/redux-toolkit/overview

It uses immer under the hood.